fix_proto_conf_desc.py

```diff
@@ -0,0 +1,177 @@
+import ida_segment
+import idautils
+import idc
+import ida_bytes
+import ida_name
+from ida_nalt import REF_OFF32, REFINFO_SIGNEDOP, REFINFO_SELFREF
+from idc import FF_DATA, FF_DWORD
+
+class CStruct: 
+    tif = 0
+    SELFREF = REFINFO_SIGNEDOP | REFINFO_SELFREF
+
+    def append_int16_member(self, name):
+        udm = ida_typeinf.udm_t()
```

Comment:
this code duplication still persists. i suggest the following:
```python
def _append_member(self, name: str, ida_type: int) -> ida_typeinf.udm_t:
    """
    Append a udm_t into `self.tif` and return it so user is able to add his
    own modifications.
    """
    udm = ida_typeinf.udm_t()
    udm.name = name
    udm.offset = self.tif.get_unpadded_size() * 8
    udm.type = ida_typeinf.tinfo_t(ida_type)
    udm.size = udm.type.get_size() * 8
    self.tif.add_udm(udm)
    return udm
```

And since `udm` is now returned, you could use it as follows:

```python
def append_reloffset_member(self, name: str) -> None:
    udm = self._append_member(name, ida_typeinf.BT_INT32)
    udm.repr.set_vtype(ida_typeinf.FRB_OFFSET)
    udm.repr.ri.init(ida_nalt.REF_OFF32 | self.SELFREF)
````

Reply:
That's such a great idea! Will apply the change!

Reply:
Just implemented it and tested it. You have to set the udm type to offset and them add it to the tif. If you add the udm to the tif and after define as offset it won't work. It's the same idea but with that different detail :)